src/utils/jsonPointers.ts

```diff
@@ -2,69 +2,36 @@ import { syntaxTree } from "@codemirror/language";
 import { EditorState, Text } from "@codemirror/state";
 import { SyntaxNode, SyntaxNodeRef } from "@lezer/common";
 import { JSONPointersMap, Side } from "../types";
-
-const VAL_NODE_NAME = /^(?:Null|True|False|Object|Array|String|Number)$/;
+import { TOKENS } from "../constants";
+import { findNodeIndexInArrayNode, getWord, isValueNode } from "./node";
 
 export type JSONMode = "json4" | "json5";
 
-// borrowed from `codemirror-json5`
-// TODO: determine from spec if {"prop'name": example} is valid in json5
-// JSON5 parse doesn't support parsing individual strings like JSON.parse does
-
-function json5PropNameParser(s: string) {
-  if (s.length < 2) return s;
-  let first = s[0];
-  let last = s[s.length - 1];
-  if ((first === `'` && last === `'`) || (first === `"` && last === `"`)) {
-    s = s.slice(1, -1);
-  }
-  return s;
-}
-
-const propNameParsers: Record<JSONMode, typeof JSON.parse> = {
-  json4: JSON.parse,
-  json5: json5PropNameParser,
-};
-
-/**
- * get a JSON4/5 pointer for a given node in the editor
- *
- * adapted from https://discuss.codemirror.net/t/json-pointer-at-cursor-seeking-implementation-critique/4793/3
- * this could be useful for other things later!
- * @group Utilities
- * @internal
- */
+// adapted from https://discuss.codemirror.net/t/json-pointer-at-cursor-seeking-implementation-critique/4793/3
+// this could be useful for other things later!
 export function getJsonPointerAt(
   docText: Text,
   node: SyntaxNode,
   mode: JSONMode = "json4"
```

Comment:
it looks like we can drop `mode` altogether now - I need to write more comprehensive tests for json5, but I think that this pointer utility now works for either json4 or json5 spec?

Reply:
Yes